dht/range_streamer.hh

```diff
@@ -158,6 +161,7 @@ private:
     sstring _description;
     streaming::stream_reason _reason;
     std::vector<sstring> _tables;
+    service::frozen_topology_guard _topo_guard;
```

Comment:
So the range streamer hold only a reference to the guard, but why? The guard needs to exists all the time the streaming is running, no? How do we guaranty it?

Reply:
It doesn't have to exist all the time, only around section which has side-effects. If the coordinator moves on while the guard is not materialized, streaming will fail later when it tries to materialize the guard. It's even more efficient to not materialize early, because it won't block the coordinator unnecessarily.